### PR TITLE
v4.0.7

### PR DESCRIPTION
**Pull request by Snowflake107**
## Changes
Bugfixes for:
#402
#318
#391
and others

## Status

- [x] These changes have been tested and formatted properly.
- [ ] This PR includes only documentation changes, no code change.
- [ ] This PR introduces some Breaking changes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,6 +65,7 @@
     "@discordjs/opus": "^0.5.0",
     "@types/node": "^14.14.41",
     "@types/ws": "^7.4.1",
+    "discord-api-types": "^0.18.1",
```
Comment (Androz2091) [inline package.json:68]: What's the point of adding this dependency? Seems like it's not used.

**Review by Androz2091 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Snowflake107) [reply package.json:68]: it caused some TypeScript error xD 🙄

Comment (Snowflake107) [reply package.json]: looks like an issue with the language server

**Review by Androz2091 — APPROVED** [review]
LGTM 👍 
Can be merged.

File: package.json (hunk 1 above)
Comment (Androz2091) [reply package.json]: What do you mean? 🤔 
It compiles well without d-api-types btw ✅

**Pull request merged**